{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "ename": "OSError",
     "evalue": "[E050] Can't find model 'en_core_web_sm'. It doesn't seem to be a shortcut link, a Python package or a valid path to a data directory.",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mOSError\u001b[0m                                   Traceback (most recent call last)",
      "\u001b[1;32mc:\\Users\\Apprenant\\Desktop\\Dev IA\\Emotion_app\\ML\\scikit_learn_rudy.ipynb Cell 1'\u001b[0m in \u001b[0;36m<cell line: 5>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      <a href='vscode-notebook-cell:/c%3A/Users/Apprenant/Desktop/Dev%20IA/Emotion_app/ML/scikit_learn_rudy.ipynb#ch0000000?line=2'>3</a>\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39msklearn\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mfeature_extraction\u001b[39;00m\u001b[39m.\u001b[39;00m\u001b[39mtext\u001b[39;00m \u001b[39mimport\u001b[39;00m CountVectorizer\n\u001b[0;32m      <a href='vscode-notebook-cell:/c%3A/Users/Apprenant/Desktop/Dev%20IA/Emotion_app/ML/scikit_learn_rudy.ipynb#ch0000000?line=3'>4</a>\u001b[0m \u001b[39mimport\u001b[39;00m \u001b[39mspacy\u001b[39;00m\n\u001b[1;32m----> <a href='vscode-notebook-cell:/c%3A/Users/Apprenant/Desktop/Dev%20IA/Emotion_app/ML/scikit_learn_rudy.ipynb#ch0000000?line=4'>5</a>\u001b[0m spacy\u001b[39m.\u001b[39;49mload(\u001b[39m'\u001b[39;49m\u001b[39men_core_web_sm\u001b[39;49m\u001b[39m'\u001b[39;49m)\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\spacy\\__init__.py:30\u001b[0m, in \u001b[0;36mload\u001b[1;34m(name, **overrides)\u001b[0m\n\u001b[0;32m     <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/spacy/__init__.py?line=27'>28</a>\u001b[0m \u001b[39mif\u001b[39;00m depr_path \u001b[39mnot\u001b[39;00m \u001b[39min\u001b[39;00m (\u001b[39mTrue\u001b[39;00m, \u001b[39mFalse\u001b[39;00m, \u001b[39mNone\u001b[39;00m):\n\u001b[0;32m     <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/spacy/__init__.py?line=28'>29</a>\u001b[0m     warnings\u001b[39m.\u001b[39mwarn(Warnings\u001b[39m.\u001b[39mW001\u001b[39m.\u001b[39mformat(path\u001b[39m=\u001b[39mdepr_path), \u001b[39mDeprecationWarning\u001b[39;00m)\n\u001b[1;32m---> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/spacy/__init__.py?line=29'>30</a>\u001b[0m \u001b[39mreturn\u001b[39;00m util\u001b[39m.\u001b[39;49mload_model(name, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49moverrides)\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\spacy\\util.py:175\u001b[0m, in \u001b[0;36mload_model\u001b[1;34m(name, **overrides)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/spacy/util.py?line=172'>173</a>\u001b[0m \u001b[39melif\u001b[39;00m \u001b[39mhasattr\u001b[39m(name, \u001b[39m\"\u001b[39m\u001b[39mexists\u001b[39m\u001b[39m\"\u001b[39m):  \u001b[39m# Path or Path-like to model data\u001b[39;00m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/spacy/util.py?line=173'>174</a>\u001b[0m     \u001b[39mreturn\u001b[39;00m load_model_from_path(name, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39moverrides)\n\u001b[1;32m--> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/spacy/util.py?line=174'>175</a>\u001b[0m \u001b[39mraise\u001b[39;00m \u001b[39mIOError\u001b[39;00m(Errors\u001b[39m.\u001b[39mE050\u001b[39m.\u001b[39mformat(name\u001b[39m=\u001b[39mname))\n",
      "\u001b[1;31mOSError\u001b[0m: [E050] Can't find model 'en_core_web_sm'. It doesn't seem to be a shortcut link, a Python package or a valid path to a data directory."
     ]
    }
   ],
   "source": [
    "from pycaret.nlp import *\n",
    "import pandas as pd\n",
    "from sklearn.feature_extraction.text import CountVectorizer\n",
    "import spacy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfe = pd.read_csv(\"../Data/Data_cleaned/emotion_cleaned_rudy.csv\")[[\"target\", \"clean_text\"]].dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CountVectorizer()"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vectorizer = CountVectorizer()\n",
    "vectorizer.fit(dfe[\"clean_text\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# data = vectorizer.transform(dfe[\"clean_text\"])\n",
    "# df = pd.DataFrame(data)\n",
    "# df[\"target\"] = dfe.target"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "ename": "AssertionError",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mOSError\u001b[0m                                   Traceback (most recent call last)",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\pycaret\\nlp.py:313\u001b[0m, in \u001b[0;36msetup\u001b[1;34m(data, target, custom_stopwords, html, session_id, log_experiment, experiment_name, log_plots, log_data, verbose)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/pycaret/nlp.py?line=310'>311</a>\u001b[0m     \u001b[39mimport\u001b[39;00m \u001b[39mspacy\u001b[39;00m\n\u001b[1;32m--> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/pycaret/nlp.py?line=312'>313</a>\u001b[0m     sp \u001b[39m=\u001b[39m spacy\u001b[39m.\u001b[39;49mload(\u001b[39m\"\u001b[39;49m\u001b[39men_core_web_sm\u001b[39;49m\u001b[39m\"\u001b[39;49m, disable\u001b[39m=\u001b[39;49m[\u001b[39m\"\u001b[39;49m\u001b[39mparser\u001b[39;49m\u001b[39m\"\u001b[39;49m, \u001b[39m\"\u001b[39;49m\u001b[39mner\u001b[39;49m\u001b[39m\"\u001b[39;49m])\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/pycaret/nlp.py?line=313'>314</a>\u001b[0m \u001b[39mexcept\u001b[39;00m:\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\spacy\\__init__.py:30\u001b[0m, in \u001b[0;36mload\u001b[1;34m(name, **overrides)\u001b[0m\n\u001b[0;32m     <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/spacy/__init__.py?line=28'>29</a>\u001b[0m     warnings\u001b[39m.\u001b[39mwarn(Warnings\u001b[39m.\u001b[39mW001\u001b[39m.\u001b[39mformat(path\u001b[39m=\u001b[39mdepr_path), \u001b[39mDeprecationWarning\u001b[39;00m)\n\u001b[1;32m---> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/spacy/__init__.py?line=29'>30</a>\u001b[0m \u001b[39mreturn\u001b[39;00m util\u001b[39m.\u001b[39;49mload_model(name, \u001b[39m*\u001b[39;49m\u001b[39m*\u001b[39;49moverrides)\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\spacy\\util.py:175\u001b[0m, in \u001b[0;36mload_model\u001b[1;34m(name, **overrides)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/spacy/util.py?line=173'>174</a>\u001b[0m     \u001b[39mreturn\u001b[39;00m load_model_from_path(name, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39moverrides)\n\u001b[1;32m--> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/spacy/util.py?line=174'>175</a>\u001b[0m \u001b[39mraise\u001b[39;00m \u001b[39mIOError\u001b[39;00m(Errors\u001b[39m.\u001b[39mE050\u001b[39m.\u001b[39mformat(name\u001b[39m=\u001b[39mname))\n",
      "\u001b[1;31mOSError\u001b[0m: [E050] Can't find model 'en_core_web_sm'. It doesn't seem to be a shortcut link, a Python package or a valid path to a data directory.",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mSystemExit\u001b[0m                                Traceback (most recent call last)",
      "    \u001b[1;31m[... skipping hidden 1 frame]\u001b[0m\n",
      "\u001b[1;32mc:\\Users\\Apprenant\\Desktop\\Dev IA\\Emotion_app\\ML\\scikit_learn_rudy.ipynb Cell 5'\u001b[0m in \u001b[0;36m<cell line: 1>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> <a href='vscode-notebook-cell:/c%3A/Users/Apprenant/Desktop/Dev%20IA/Emotion_app/ML/scikit_learn_rudy.ipynb#ch0000002?line=0'>1</a>\u001b[0m s \u001b[39m=\u001b[39m setup(dfe, target\u001b[39m=\u001b[39;49m\u001b[39m\"\u001b[39;49m\u001b[39mtarget\u001b[39;49m\u001b[39m\"\u001b[39;49m)\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\pycaret\\nlp.py:315\u001b[0m, in \u001b[0;36msetup\u001b[1;34m(data, target, custom_stopwords, html, session_id, log_experiment, experiment_name, log_plots, log_data, verbose)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/pycaret/nlp.py?line=313'>314</a>\u001b[0m \u001b[39mexcept\u001b[39;00m:\n\u001b[1;32m--> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/pycaret/nlp.py?line=314'>315</a>\u001b[0m     sys\u001b[39m.\u001b[39;49mexit(\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/pycaret/nlp.py?line=315'>316</a>\u001b[0m         \u001b[39m\"\u001b[39;49m\u001b[39m(Type Error): spacy english model is not yet downloaded. See the documentation of setup to see installation guide.\u001b[39;49m\u001b[39m\"\u001b[39;49m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/pycaret/nlp.py?line=316'>317</a>\u001b[0m     )\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/pycaret/nlp.py?line=318'>319</a>\u001b[0m \u001b[39m# html\u001b[39;00m\n",
      "\u001b[1;31mSystemExit\u001b[0m: (Type Error): spacy english model is not yet downloaded. See the documentation of setup to see installation guide.",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[1;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "    \u001b[1;31m[... skipping hidden 1 frame]\u001b[0m\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\IPython\\core\\interactiveshell.py:1972\u001b[0m, in \u001b[0;36mInteractiveShell.showtraceback\u001b[1;34m(self, exc_tuple, filename, tb_offset, exception_only, running_compiled_code)\u001b[0m\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/interactiveshell.py?line=1968'>1969</a>\u001b[0m \u001b[39mif\u001b[39;00m exception_only:\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/interactiveshell.py?line=1969'>1970</a>\u001b[0m     stb \u001b[39m=\u001b[39m [\u001b[39m'\u001b[39m\u001b[39mAn exception has occurred, use \u001b[39m\u001b[39m%\u001b[39m\u001b[39mtb to see \u001b[39m\u001b[39m'\u001b[39m\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/interactiveshell.py?line=1970'>1971</a>\u001b[0m            \u001b[39m'\u001b[39m\u001b[39mthe full traceback.\u001b[39m\u001b[39m\\n\u001b[39;00m\u001b[39m'\u001b[39m]\n\u001b[1;32m-> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/interactiveshell.py?line=1971'>1972</a>\u001b[0m     stb\u001b[39m.\u001b[39mextend(\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mInteractiveTB\u001b[39m.\u001b[39;49mget_exception_only(etype,\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/interactiveshell.py?line=1972'>1973</a>\u001b[0m                                                      value))\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/interactiveshell.py?line=1973'>1974</a>\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/interactiveshell.py?line=1974'>1975</a>\u001b[0m     \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/interactiveshell.py?line=1975'>1976</a>\u001b[0m         \u001b[39m# Exception classes can customise their traceback - we\u001b[39;00m\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/interactiveshell.py?line=1976'>1977</a>\u001b[0m         \u001b[39m# use this in IPython.parallel for exceptions occurring\u001b[39;00m\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/interactiveshell.py?line=1977'>1978</a>\u001b[0m         \u001b[39m# in the engines. This should return a list of strings.\u001b[39;00m\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\IPython\\core\\ultratb.py:578\u001b[0m, in \u001b[0;36mListTB.get_exception_only\u001b[1;34m(self, etype, value)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=569'>570</a>\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mget_exception_only\u001b[39m(\u001b[39mself\u001b[39m, etype, value):\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=570'>571</a>\u001b[0m     \u001b[39m\"\"\"Only print the exception type and message, without a traceback.\u001b[39;00m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=571'>572</a>\u001b[0m \n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=572'>573</a>\u001b[0m \u001b[39m    Parameters\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=575'>576</a>\u001b[0m \u001b[39m    value : exception value\u001b[39;00m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=576'>577</a>\u001b[0m \u001b[39m    \"\"\"\u001b[39;00m\n\u001b[1;32m--> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=577'>578</a>\u001b[0m     \u001b[39mreturn\u001b[39;00m ListTB\u001b[39m.\u001b[39;49mstructured_traceback(\u001b[39mself\u001b[39;49m, etype, value)\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\IPython\\core\\ultratb.py:436\u001b[0m, in \u001b[0;36mListTB.structured_traceback\u001b[1;34m(self, etype, evalue, etb, tb_offset, context)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=432'>433</a>\u001b[0m     chained_exc_ids\u001b[39m.\u001b[39madd(\u001b[39mid\u001b[39m(exception[\u001b[39m1\u001b[39m]))\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=433'>434</a>\u001b[0m     chained_exceptions_tb_offset \u001b[39m=\u001b[39m \u001b[39m0\u001b[39m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=434'>435</a>\u001b[0m     out_list \u001b[39m=\u001b[39m (\n\u001b[1;32m--> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=435'>436</a>\u001b[0m         \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mstructured_traceback(\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=436'>437</a>\u001b[0m             etype, evalue, (etb, chained_exc_ids),\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=437'>438</a>\u001b[0m             chained_exceptions_tb_offset, context)\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=438'>439</a>\u001b[0m         \u001b[39m+\u001b[39m chained_exception_message\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=439'>440</a>\u001b[0m         \u001b[39m+\u001b[39m out_list)\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=441'>442</a>\u001b[0m \u001b[39mreturn\u001b[39;00m out_list\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\IPython\\core\\ultratb.py:1105\u001b[0m, in \u001b[0;36mAutoFormattedTB.structured_traceback\u001b[1;34m(self, etype, value, tb, tb_offset, number_of_lines_of_context)\u001b[0m\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=1102'>1103</a>\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=1103'>1104</a>\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mtb \u001b[39m=\u001b[39m tb\n\u001b[1;32m-> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=1104'>1105</a>\u001b[0m \u001b[39mreturn\u001b[39;00m FormattedTB\u001b[39m.\u001b[39;49mstructured_traceback(\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=1105'>1106</a>\u001b[0m     \u001b[39mself\u001b[39;49m, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\IPython\\core\\ultratb.py:999\u001b[0m, in \u001b[0;36mFormattedTB.structured_traceback\u001b[1;34m(self, etype, value, tb, tb_offset, number_of_lines_of_context)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=995'>996</a>\u001b[0m mode \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mmode\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=996'>997</a>\u001b[0m \u001b[39mif\u001b[39;00m mode \u001b[39min\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mverbose_modes:\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=997'>998</a>\u001b[0m     \u001b[39m# Verbose modes need a full traceback\u001b[39;00m\n\u001b[1;32m--> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=998'>999</a>\u001b[0m     \u001b[39mreturn\u001b[39;00m VerboseTB\u001b[39m.\u001b[39;49mstructured_traceback(\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=999'>1000</a>\u001b[0m         \u001b[39mself\u001b[39;49m, etype, value, tb, tb_offset, number_of_lines_of_context\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=1000'>1001</a>\u001b[0m     )\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=1001'>1002</a>\u001b[0m \u001b[39melif\u001b[39;00m mode \u001b[39m==\u001b[39m \u001b[39m'\u001b[39m\u001b[39mMinimal\u001b[39m\u001b[39m'\u001b[39m:\n\u001b[0;32m   <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=1002'>1003</a>\u001b[0m     \u001b[39mreturn\u001b[39;00m ListTB\u001b[39m.\u001b[39mget_exception_only(\u001b[39mself\u001b[39m, etype, value)\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\IPython\\core\\ultratb.py:852\u001b[0m, in \u001b[0;36mVerboseTB.structured_traceback\u001b[1;34m(self, etype, evalue, etb, tb_offset, number_of_lines_of_context)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=849'>850</a>\u001b[0m \u001b[39m\"\"\"Return a nice text document describing the traceback.\"\"\"\u001b[39;00m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=850'>851</a>\u001b[0m \u001b[39massert\u001b[39;00m etb \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m\n\u001b[1;32m--> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=851'>852</a>\u001b[0m formatted_exception \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mformat_exception_as_a_whole(etype, evalue, etb, number_of_lines_of_context,\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=852'>853</a>\u001b[0m                                                        tb_offset)\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=854'>855</a>\u001b[0m colors \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mColors  \u001b[39m# just a shorthand + quicker name lookup\u001b[39;00m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=855'>856</a>\u001b[0m colorsnormal \u001b[39m=\u001b[39m colors\u001b[39m.\u001b[39mNormal  \u001b[39m# used a lot\u001b[39;00m\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\IPython\\core\\ultratb.py:786\u001b[0m, in \u001b[0;36mVerboseTB.format_exception_as_a_whole\u001b[1;34m(self, etype, evalue, etb, number_of_lines_of_context, tb_offset)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=783'>784</a>\u001b[0m \u001b[39massert\u001b[39;00m \u001b[39misinstance\u001b[39m(tb_offset, \u001b[39mint\u001b[39m)\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=784'>785</a>\u001b[0m head \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mprepare_header(etype, \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mlong_header)\n\u001b[1;32m--> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=785'>786</a>\u001b[0m records \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mget_records(etb, number_of_lines_of_context, tb_offset)\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=787'>788</a>\u001b[0m frames \u001b[39m=\u001b[39m []\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=788'>789</a>\u001b[0m skipped \u001b[39m=\u001b[39m \u001b[39m0\u001b[39m\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\IPython\\core\\ultratb.py:840\u001b[0m, in \u001b[0;36mVerboseTB.get_records\u001b[1;34m(self, etb, number_of_lines_of_context, tb_offset)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=833'>834</a>\u001b[0m options \u001b[39m=\u001b[39m stack_data\u001b[39m.\u001b[39mOptions(\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=834'>835</a>\u001b[0m     before\u001b[39m=\u001b[39mbefore,\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=835'>836</a>\u001b[0m     after\u001b[39m=\u001b[39mafter,\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=836'>837</a>\u001b[0m     pygments_formatter\u001b[39m=\u001b[39mformatter,\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=837'>838</a>\u001b[0m )\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=838'>839</a>\u001b[0m \u001b[39massert\u001b[39;00m etb \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m\n\u001b[1;32m--> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/IPython/core/ultratb.py?line=839'>840</a>\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mlist\u001b[39;49m(stack_data\u001b[39m.\u001b[39;49mFrameInfo\u001b[39m.\u001b[39;49mstack_data(etb, options\u001b[39m=\u001b[39;49moptions))[tb_offset:]\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\stack_data\\core.py:546\u001b[0m, in \u001b[0;36mFrameInfo.stack_data\u001b[1;34m(cls, frame_or_tb, options, collapse_repeated_frames)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=529'>530</a>\u001b[0m \u001b[39m@classmethod\u001b[39m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=530'>531</a>\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mstack_data\u001b[39m(\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=531'>532</a>\u001b[0m         \u001b[39mcls\u001b[39m,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=535'>536</a>\u001b[0m         collapse_repeated_frames: \u001b[39mbool\u001b[39m \u001b[39m=\u001b[39m \u001b[39mTrue\u001b[39;00m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=536'>537</a>\u001b[0m ) \u001b[39m-\u001b[39m\u001b[39m>\u001b[39m Iterator[Union[\u001b[39m'\u001b[39m\u001b[39mFrameInfo\u001b[39m\u001b[39m'\u001b[39m, RepeatedFrames]]:\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=537'>538</a>\u001b[0m     \u001b[39m\"\"\"\u001b[39;00m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=538'>539</a>\u001b[0m \u001b[39m    An iterator of FrameInfo and RepeatedFrames objects representing\u001b[39;00m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=539'>540</a>\u001b[0m \u001b[39m    a full traceback or stack. Similar consecutive frames are collapsed into RepeatedFrames\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=543'>544</a>\u001b[0m \u001b[39m    and optionally an Options object to configure.\u001b[39;00m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=544'>545</a>\u001b[0m \u001b[39m    \"\"\"\u001b[39;00m\n\u001b[1;32m--> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=545'>546</a>\u001b[0m     stack \u001b[39m=\u001b[39m \u001b[39mlist\u001b[39;49m(iter_stack(frame_or_tb))\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=547'>548</a>\u001b[0m     \u001b[39m# Reverse the stack from a frame so that it's in the same order\u001b[39;00m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=548'>549</a>\u001b[0m     \u001b[39m# as the order from a traceback, which is the order of a printed\u001b[39;00m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=549'>550</a>\u001b[0m     \u001b[39m# traceback when read top to bottom (most recent call last)\u001b[39;00m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/core.py?line=550'>551</a>\u001b[0m     \u001b[39mif\u001b[39;00m is_frame(frame_or_tb):\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\stack_data\\utils.py:98\u001b[0m, in \u001b[0;36miter_stack\u001b[1;34m(frame_or_tb)\u001b[0m\n\u001b[0;32m     <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/utils.py?line=95'>96</a>\u001b[0m \u001b[39mwhile\u001b[39;00m frame_or_tb:\n\u001b[0;32m     <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/utils.py?line=96'>97</a>\u001b[0m     \u001b[39myield\u001b[39;00m frame_or_tb\n\u001b[1;32m---> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/utils.py?line=97'>98</a>\u001b[0m     \u001b[39mif\u001b[39;00m is_frame(frame_or_tb):\n\u001b[0;32m     <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/utils.py?line=98'>99</a>\u001b[0m         frame_or_tb \u001b[39m=\u001b[39m frame_or_tb\u001b[39m.\u001b[39mf_back\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/utils.py?line=99'>100</a>\u001b[0m     \u001b[39melse\u001b[39;00m:\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\stack_data\\utils.py:91\u001b[0m, in \u001b[0;36mis_frame\u001b[1;34m(frame_or_tb)\u001b[0m\n\u001b[0;32m     <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/utils.py?line=89'>90</a>\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mis_frame\u001b[39m(frame_or_tb: Union[FrameType, TracebackType]) \u001b[39m-\u001b[39m\u001b[39m>\u001b[39m \u001b[39mbool\u001b[39m:\n\u001b[1;32m---> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/utils.py?line=90'>91</a>\u001b[0m     assert_(\u001b[39misinstance\u001b[39;49m(frame_or_tb, (types\u001b[39m.\u001b[39;49mFrameType, types\u001b[39m.\u001b[39;49mTracebackType)))\n\u001b[0;32m     <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/utils.py?line=91'>92</a>\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39misinstance\u001b[39m(frame_or_tb, (types\u001b[39m.\u001b[39mFrameType,))\n",
      "File \u001b[1;32m~\\anaconda3\\envs\\pycaret0\\lib\\site-packages\\stack_data\\utils.py:172\u001b[0m, in \u001b[0;36massert_\u001b[1;34m(condition, error)\u001b[0m\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/utils.py?line=169'>170</a>\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39misinstance\u001b[39m(error, \u001b[39mstr\u001b[39m):\n\u001b[0;32m    <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/utils.py?line=170'>171</a>\u001b[0m     error \u001b[39m=\u001b[39m \u001b[39mAssertionError\u001b[39;00m(error)\n\u001b[1;32m--> <a href='file:///c%3A/Users/Apprenant/anaconda3/envs/pycaret0/lib/site-packages/stack_data/utils.py?line=171'>172</a>\u001b[0m \u001b[39mraise\u001b[39;00m error\n",
      "\u001b[1;31mAssertionError\u001b[0m: "
     ]
    }
   ],
   "source": [
    "s = setup(dfe, target=\"target\")"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "c13c1cfe5359d1bee369f00506b618d79613c29c373f6c1da3b782ba511a04fb"
  },
  "kernelspec": {
   "display_name": "Python 3.8.0 ('pycaret0')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.0"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
